corehq/apps/campaign/models.py

```diff
@@ -95,3 +137,17 @@ class DashboardGauge(DashboardWidgetBase):
 
     # optional additional configuration set to customize gauge appearance
     configuration = JSONField(default=dict)
+
+
+def model_to_widget(instance, fields=None, exclude=None, properties=()):
+    """
+    Like model_to_dict, but adds 'widget_type', 'dashboard', and
+    properties given in ``properties``.
+    """
+    widget = model_to_dict(instance, fields, exclude)
+    widget.update(
+        {prop: getattr(instance, prop) for prop in properties},
+        widget_type=instance.__class__.__name__,
+        dashboard=model_to_dict(instance.dashboard, exclude=['id']),
```

Comment:
We (or at least I) will need the widget's dashboard to get to the report's domain. Django's `model_to_dict()` sets the widget's "dashboard" property to its ID. This line sets it to a dictionary instead.